libs/api-services/src/lib/session/extendSession.ts

```diff
@@ -0,0 +1,49 @@
+import { prisma } from '@dnd-assistant/prisma';
+import { generateSessionExpiry } from './generate-session-expiry';
+import { Session } from '@prisma/client';
+
+/**
+ * We will only extend a session if it expires within this many days
+ */
+const EXTEND_IF_EXPIRES_WITHIN_DAYS = 7;
+/**
+ * When extending a session, it will be valid for this length
+ */
+const EXTENDED_SESSION_VALID_DAYS = 60;
+
+/**
+ * Attempts to extend session if possible, otherwise returns early
+ * Only throws on a DB error
+ */
+export const extendSession = async (session: Session) => {
+  if (new Date(session.expiresAt) < new Date()) {
+    // Session has already expired, cannot be extended
+    return;
+  }
+
+  const extensionWindow = new Date();
+  extensionWindow.setDate(new Date().getDate() + EXTEND_IF_EXPIRES_WITHIN_DAYS);
```

Comment:
can make use of `generateSessionExpiry `here

Reply:
Ah whoopsie, that's why I added the argument to generateSessionExpiry in the first place, then promptly forgot to use it...

Reply:
Although that does decrease linecount, I believe it may make the code more confusing, since here we're really not generating a session expiry. I worry if we ever add additional logic to generateSessionExpiry, it really doesn't apply 1:1 here since we're not generating a session expiry here.